release/internal/pinnedversion/pinnedversion.go

```diff
@@ -211,9 +221,14 @@ func GenerateOperatorComponents(srcDir, outputDir string) (registry.OperatorComp
 	if err != nil {
 		return op, "", err
 	}
-	for name, component := range pinnedVersion.Components {
-		pinnedVersion.Components[name] = normalizeComponent(name, component)
+	// Remove components that are not needed in the operator components file.
+	// These either do not produce images or are not used by the operator.
+	components := pinnedVersion.Components
+	for _, c := range operatorIgnoreComponents {
+		delete(components, c)
 	}
+	pinnedVersion.Components = components
```

Comment:
```suggestion
	for _, c := range operatorIgnoreComponents {
		delete(pinnedVersion.Components, c)
}
```

Do we need the intermediate variable here?